From a634369dbe96281a91b6c38ddf1e6995d4505365 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Tue, 20 Sep 2016 14:52:58 +0200
Subject: [PATCH 044/296] Handle IllegalArgumentException

In some cases startInputInnerLocked() throws
IllegalArgumentException which needs to be handle
---
 .../core/java/com/android/server/InputMethodManagerService.java    | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/InputMethodManagerService.java b/services/core/java/com/android/server/InputMethodManagerService.java
index d177e596cb0..c5feaebdf16 100644
--- a/services/core/java/com/android/server/InputMethodManagerService.java
+++ b/services/core/java/com/android/server/InputMethodManagerService.java
@@ -1865,7 +1865,12 @@ public class InputMethodManagerService extends IInputMethodManager.Stub
             }
         }
 
-        return startInputInnerLocked();
+        try {
+            return startInputInnerLocked();
+        } catch (RuntimeException e) {
+            Slog.w(TAG, "Unexpected exception", e);
+            return null;
+        }
     }
 
     InputBindResult startInputInnerLocked() {
-- 
2.11.0

